lib/addons/aws-batch-on-eks/index.ts

```diff
@@ -0,0 +1,38 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { Stack } from "aws-cdk-lib";
+import { CfnServiceLinkedRole, IRole, Role } from "aws-cdk-lib/aws-iam";
+import { Construct } from "constructs";
+
+const BATCH = 'aws-batch';
+
+export class AwsBatchAddOn implements ClusterAddOn {
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const cluster = clusterInfo.cluster;
+
+    // Create the service role used by AWS Batch on EKS 
+    const batchOnEksSlr = new CfnServiceLinkedRole(cluster.stack, 'BatchServiceRole', {
```

Comment:
This role is created once per account. If it is already created, will this fail? I was not able to tell by looking at `CfnServiceLinkedRole` CDK docs, it appears that it may fail. As we discussed, if the role is only expected to be created once per account, we could supply a flag to control this behavior.